pkg/chartutil/create.go

```diff
@@ -290,6 +292,26 @@ Create chart name and version as used by the chart label.
 {{- end -}}
 `
 
+const defaultServiceTest = `apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ template "<CHARTNAME>.fullname" . }}-service-test"
+  labels:
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app: {{ template "<CHARTNAME>.name" . }}
```

Comment:
Labels should match those of other resources and `include` should be preferred over `template`:

```yaml
app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . }}
helm.sh/chart: {{ include "<CHARTNAME>.chart" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
```

Reply:
Good catch. Updated.